contracts/DEXes/UniV3StablesJoint.sol

```diff
@@ -0,0 +1,639 @@
+                _minTick,
+                _maxTick,
+                _pool.feeGrowthGlobal0X128(),
+                _pool.feeGrowthGlobal1X128(),
+                positionInfo.feeGrowthInside0LastX128,
+                positionInfo.feeGrowthInside1LastX128
+            ),
+            _pool.ticks(_minTick),
+            _pool.ticks(_maxTick)
+        );
+
+        // Reorder to make sure amounts are added correctly
+        if (tokenA < tokenB) {
+            _amountPending[0] += tokensOwed0;
+            _amountPending[1] += tokensOwed1;
+        } else {
+            _amountPending[1] += tokensOwed0;
+            _amountPending[0] += tokensOwed1;
+        }
+
+        return _amountPending;
+    }
+
+    /*
+     * @notice
+     *  Function called by the uniswap pool when minting the LP position (providing liquidity),
+     * instead of approving and sending the tokens, uniV3 calls the callback imoplementation
+     * on the caller contract
+     * @param amount0Owed, amount of token0 to send
+     * @param amount1Owed, amount of token1 to send
+     * @param data, additional calldata
+     */
+    function uniswapV3MintCallback(
+        uint256 amount0Owed,
+        uint256 amount1Owed,
+        bytes calldata data
+    ) external {
+        // Only the pool can use this function
+        require(msg.sender == pool); // dev: callback only called by pool
+        // Send the required funds to the pool
+        IUniswapV3Pool _pool = IUniswapV3Pool(pool);
+        IERC20(_pool.token0()).safeTransfer(address(_pool), amount0Owed);
+        IERC20(_pool.token1()).safeTransfer(address(_pool), amount1Owed);
+    }
+
+    /*
+     * @notice
+     *  Function called by the uniswap pool when swapping,
+     * instead of approving and sending the tokens, uniV3 calls the callback imoplementation
+     * on the caller contract
+     * @param amount0Delta, amount of token0 to send (if any)
+     * @param amount1Delta, amount of token1 to send (if any)
+     * @param data, additional calldata
+     */
+    function uniswapV3SwapCallback(
+        int256 amount0Delta,
+        int256 amount1Delta,
+        bytes calldata data
+    ) external {
+        // Only the pool can use this function
+        require(msg.sender == address(pool)); // dev: callback only called by pool
+
+        IUniswapV3Pool _pool = IUniswapV3Pool(pool);
+
+        uint256 amountIn;
+        address tokenIn;
+
+        // Send the required funds to the pool
+        if (amount0Delta > 0) {
+            amountIn = uint256(amount0Delta);
+            tokenIn = _pool.token0();
+        } else {
+            amountIn = uint256(amount1Delta);
+            tokenIn = _pool.token1();
+        }
+
+        IERC20(tokenIn).safeTransfer(address(_pool), amountIn);
+    }
+
+    /*
+     * @notice
+     *  Function claiming the earned rewards for the joint, sends the tokens to the joint
+     * contract
+     */
+    function getReward() internal override {
+        IUniswapV3Pool(pool).collect(
```

Comment:
do you need to update the position first so fees are accumulated in tokensOwed0 and 1?

Reply:
Ideally yes but I have not found an external function in Uni to update it so the function is always used after a position updating call (burnLP for example) and we need to include a burnLP(0) (does not revert, already tested) in the call to collect rewards manually. I propose also to change the name to `collectOwedTokens` as this is used both to collect rewards and principal